harness/determined/deploy/local/cluster_utils.py

```diff
@@ -271,6 +277,23 @@ def agent_up(
     cluster_name: str,
     labels: Optional[Dict] = None,
 ) -> None:
+    agent_conf = {}
+    volumes = ["/var/run/docker.sock:/var/run/docker.sock"]
+    if agent_config_path is not None:
+        with agent_config_path.open() as f:
+            agent_conf = yaml.safe_load(f)
+        volumes += [f"{agent_config_path}:/etc/determined/agent.yaml"]
+
+    # Fallback on agent config for options not specified as flags.
+    if agent_name == AGENT_NAME_DEFAULT:
+        agent_name = agent_conf.get("agent_id", f"det-agent-{socket.gethostname()}")
+    if master_port == MASTER_PORT_DEFAULT:
+        master_port = agent_conf.get("master_port", master_port)
+    if agent_label is None:
+        agent_label = agent_conf.get("label", None)
+    if agent_resource_pool is None:
+        agent_resource_pool = agent_conf.get("resource_pool", None)
```

Comment:
nitpick: maybe if these values are already in yaml, we can skip / omit them from `environment` dict here? 
that is, if it's in yaml, no point in passing the same value through env vars again.